presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -555,6 +568,152 @@ public void testDecimalSequence()
                                 false, true, 28999999999L, 28999999999L, false, true, true))));
     }
 
+    @Test
+    public void testSliceVarchar()
+            throws Exception
+    {
+        //dictionary
+        tester.testRoundTrip(VARCHAR, newArrayList(limit(cycle(ImmutableList.of("apple", "apple pie", "apple\uD835\uDC03", "apple\uFFFD")), 3000)),
+                BytesRange.of("apple".getBytes(), false, "apple".getBytes(), false, false));
+
+        //direct
+        tester.testRoundTrip(VARCHAR,
+                intsBetween(0, 20).stream().map(Object::toString).collect(toList()),
+                BytesValues.of(new byte[][] {"10".getBytes(), "11".getBytes()}, false),
+                BytesValues.of(new byte[][] {"10".getBytes(), "11".getBytes()}, true),
+                BytesRange.of("10".getBytes(), false, "14".getBytes(), false, false));
+
+        //direct & dictionary column with Range Filters
+        tester.testRoundTripTypes(
+                ImmutableList.of(VARCHAR, VARCHAR),
+                ImmutableList.of(intsBetween(0, 20).stream()
+                        .map(Object::toString)
+                        .collect(toList()), newArrayList(limit(cycle(ImmutableList.of("A", "B", "C")), 20))),
+                toSubfieldFilters(ImmutableMap.of(
+                        0, BytesRange.of("10".getBytes(), false, "16".getBytes(), false, true),
+                        1, BytesRange.of("A".getBytes(), false, "B".getBytes(), false, false))));
+
+        //stride dictionary
+        tester.testRoundTrip(VARCHAR, newArrayList(concat(ImmutableList.of("a"), nCopies(9999, "123"), ImmutableList.of("b"), nCopies(9999, "123"))));
+
+        //empty sequence
+        tester.testRoundTrip(VARCHAR, newArrayList(limit(cycle(""), 30_000)));
+
+        //testDwrfInvalidCheckpointsForRowGroupDictionary
```

Comment:
perhaps, clarify this comment: `// copy of AbstractOrcTester::testDwrfInvalidCheckpointsForRowGroupDictionary`; same below